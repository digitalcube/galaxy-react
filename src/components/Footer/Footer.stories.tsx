import React from 'react';
import { Meta, Story } from '@storybook/react';
import { Footer, FooterProps } from './Footer';
import {
  DropdownItem,
  DropdownMenu,
  DropdownToggle,
  UncontrolledButtonDropdown,
} from 'reactstrap';

const meta: Meta = {
  title: 'Components/Footer',
  component: Footer,
  parameters: {
    controls: { expanded: true },
  },
};

export default meta;

const Template: Story<FooterProps> = (args) => <Footer {...args} />;

// By passing using the Args format for exported stories, you can control the props for a component for reuse in a test
// https://storybook.js.org/docs/react/workflows/unit-testing
export const Default = Template.bind({});
Default.args = {
  serviceName: 'Shifter',
};

// By passing using the Args format for exported stories, you can control the props for a component for reuse in a test
// https://storybook.js.org/docs/react/workflows/unit-testing
export const WithLinks = Template.bind({});
WithLinks.args = {
  serviceName: 'Shifter',
  links: [
    {
      id: 'item1',
      href: '#',
      label: 'Privacy Policy',
    },
    {
      id: 'item2',
      href: '#',
      label: 'Terms of Service',
    },
  ],
};

export const WithDropdownMenu = Template.bind({});
WithDropdownMenu.args = {
  serviceName: 'Shifter',
  links: [
    {
      id: 'item1',
      href: '#',
      label: 'Privacy Policy',
    },
    {
      id: 'item2',
      href: '#',
      label: 'Terms of Service,',
    },
  ],
  children: (
    <UncontrolledButtonDropdown>
      <DropdownToggle caret outline size="sm" color="primary">
        Language
      </DropdownToggle>
      <DropdownMenu>
        <DropdownItem>japanese</DropdownItem>
        <DropdownItem>English</DropdownItem>
      </DropdownMenu>
    </UncontrolledButtonDropdown>
  ),
};
