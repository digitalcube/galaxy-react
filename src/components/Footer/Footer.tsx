import React , {FC, PropsWithChildren} from 'react'
import { Copyright } from '../Copyright/Copyright'

export type FooterProps = PropsWithChildren<{
    serviceName: string;
}>
export const Footer: FC<FooterProps> = ({serviceName, children}) => (
	<footer className="text-center">
        {children}
        <Copyright serviceName={serviceName} />
    </footer>
)