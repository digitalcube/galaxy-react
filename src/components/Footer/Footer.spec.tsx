import { render } from '@testing-library/react'
import React from 'react'
import {Footer} from './Footer'

describe('Footer', () => {
    it('to match snapshot', () => {
        const { asFragment } = render(<Footer serviceName="shifter"/>)
        expect(asFragment()).toMatchSnapshot()

    })
})