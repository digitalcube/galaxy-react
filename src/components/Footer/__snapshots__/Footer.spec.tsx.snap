// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer to match snapshot 1`] = `
<DocumentFragment>
  <footer
    class="common-footer d-flex justify-content-around align-items-center"
  >
    <div
      class="d-flex"
    />
    <p
      class="m-0 common-footer-copyright"
    >
      <small
        class="d-flex align-items-center justify-content-center"
      >
        Â©2022 shifter | A Product by 
        <a
          href="https://www.digitalcube.jp/"
          rel="nofollow noopener"
          title="DigitalCube"
        >
          <img
            alt="digitalcube"
            class="d-block"
            height="15"
            width="77"
          />
        </a>
      </small>
    </p>
    <div />
  </footer>
</DocumentFragment>
`;
